alacritty_terminal/src/term/mod.rs

```diff
@@ -320,9 +494,64 @@ impl<T> Term<T> {
             selection: None,
             cell_width: size.cell_width as usize,
             cell_height: size.cell_height as usize,
+            damage,
         }
     }
 
+    #[must_use]
+    pub fn damage(&mut self, selection: Option<SelectionRange>) -> TermDamage<'_> {
+        // Ensure that we damage entire terminal in insert mode.
+        if self.mode.contains(TermMode::INSERT) {
+            self.mark_fully_damaged();
+        }
+
+        // Add information about old cursor position and new one if they are not the same, so we
+        // cover everything that was produced by `Term::input`.
+        if self.damage.last_cursor != self.grid.cursor.point {
+            self.damage.damage_point(self.damage.last_cursor);
+        }
+
+        // Always damage current cursor.
+        self.damage_cursor();
+        self.damage.last_cursor = self.grid.cursor.point;
+
+        // Damage Vi cursor if it's present.
+        if self.mode.contains(TermMode::VI) {
+            self.damage_vi_cursor();
+        }
+
+        if self.damage.last_selection != selection {
+            let display_offset = self.grid().display_offset();
+            for selection in self.damage.last_selection.into_iter().chain(selection) {
+                self.damage.damage_selection(selection, display_offset, self.columns());
+            }
+        }
+
+        self.damage.last_selection = selection;
```

Comment:
```suggestion
        if self.damage.last_selection != selection {
            let display_offset = self.grid().display_offset();
            for selection in self.damage.last_selection.into_iter().chain(selection) {
                self.damage.damage_selection(selection, display_offset, self.columns());
            }
        }
        self.damage.last_selection = selection;
```

Just to ensure a little bit of a stronger bond between these related statements.